### PR TITLE
add `fetch` method - fixes #6

### LINKED ISSUE #6: Add possibility to fetch from network with caching
We already discussed something like this here https://github.com/sindresorhus/alfy/pull/4#issuecomment-232054583.

Here's a small wrap-up. We want to add `got` to `alfy` by default in order to make network requests. But not only that, it should be possible to cache the network requests. If later on, the same network request is performed and the cache is still valid, the cache will be returned. If the cache is not valid, a new network request is performed and the cache is updated.

Bonus feature: If the cache is not valid BUT the network request fails (user might be working offline), the cached data is still returned.

```js
alfy.fetch('https://myapi.com/foo/bar.json', {maxAge: 5000}).then(res => {
    console.log(res.body);
    // {foo: 'bar'}
});
```

Some extra notes: `fetch` is just calling `got`. So you can just pass the `method` property with value `POST` if a post request has to be performed.

@sindresorhus Did I cover everything?

### PR DESCRIPTION
**Pull request by SamVerschueren**
Added a `fetch` method with caching and fallback mechanism.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -102,6 +103,39 @@ alfy.cache = new CacheConf({
...
+	if (typeof url !== 'string') {
+		return Promise.reject(new TypeError(`Expected \`url\` to be a \`string\`, got \`${typeof url}\``));
+	}
+
+	opts = Object.assign({
+		json: true
+	}, opts);
+
+	const cachedResponse = alfy.cache.store[url] && alfy.cache.store[url].data;
+	const key = url.replace(/\./g, '\\.');
```
Comment (SamVerschueren) [inline index.js]: Had to do this for `dot-prop`. Is there a cleaner way?

File: readme.md [hunk 2]
```diff
@@ -120,6 +120,35 @@ Error or error message to be displayed.
...
+Type: `object`
+
+Any of the [got](https://github.com/sindresorhus/got#options) options.
+
+###### json
+
+Type: `boolean`<br>
+Default: `true`
+
+Parse response body with `JSON.parse` and set `accept` header to `application/json`.
```
Comment (SamVerschueren) [inline readme.md]: Copied this block from the `got` docs because I set `json` to `true` by default here.

**Comment by sindresorhus** [comment]
Could you update the readme example to use `alfy.fetch()`?

File: readme.md [hunk 3]
```diff
@@ -120,6 +120,35 @@ Error or error message to be displayed.
...
+
+##### url
+
+Type: `string`
+
+URL to be invoked.
+
+##### options
+
+Type: `object`
```
Comment (sindresorhus) [inline readme.md]: Object

File: readme.md [hunk 4]
```diff
@@ -120,6 +120,35 @@ Error or error message to be displayed.
...
 
+#### fetch(url, [options])
+
+Returns a Promise that returns the body of the response.
+
+##### url
+
+Type: `string`
+
+URL to be invoked.
```
Comment (sindresorhus) [inline readme.md]: > URL to fetch.

File: readme.md [hunk 5]
```diff
@@ -120,6 +120,35 @@ Error or error message to be displayed.
 
 <img src="media/screenshot-error.png" width="694">
 
+#### fetch(url, [options])
+
+Returns a Promise that returns the body of the response.
```
Comment (sindresorhus) [inline readme.md]: Wrap `Promise` in backticks.

File: readme.md [hunk 6]
```diff
@@ -120,6 +120,35 @@ Error or error message to be displayed.
...
+
+Type: `string`
+
+URL to be invoked.
+
+##### options
+
+Type: `object`
+
+Any of the [got](https://github.com/sindresorhus/got#options) options.
```
Comment (sindresorhus) [inline readme.md]: Wrap `got` in backticks.

File: index.js (hunk 1 above)
Comment (sindresorhus) [inline index.js]: Not that I can think of. Could hash it, but that would be any better.

**Comment by sindresorhus** [comment]
Thank again. This is awesome. I'll just to the minor changes on merge :)

**Comment by SamVerschueren** [comment]
Jeej! Super excited. Can't wait to use the new version :).

**Pull request merged**